src/scores/continuous/standard_impl.py

```diff
@@ -418,3 +419,137 @@ def pbias(
+            variability (alpha), and bias (beta) terms respectively. Defaults to (1.0, 1.0, 1.0). (see
+            equation 10 in Gupta et al. (2009) for definitions of s_rho, s_alpha and s_beta)
+        return_components: If True, the function also returns the individual terms contributing to the KGE score:
+        - `rho`: Pearson correlation coefficient between forecasts and observations.
+        - `alpha`: Variability ratio (forecast/observation standard deviation).
+        - `beta`: Bias ratio between forecast and observation mean.
+            Default is False.
+
+    Returns:
+        kge_s: The Kling-Gupta Efficiency (KGE) score as an xarray DataArray
+        If `return_components` is True, the function returns xarray DataSet kge_s with the following variables:
+        - `kge`: The KGE score.
+        - `rho`: The Pearson correlation coefficient.
+        - `alpha`: The variability ratio.
+        - `beta`: The bias term.
+
+    Notes
+    - Stats are calculated only from values for which both observations and
+      simulations are not null values.
+    - This function isn't set up to take weights.
+    - Currently this function is working only on xarray data array
+    - preserve_dims: 'all' returns NaN like pearson correlation as std dev is zero for a single point
+
+    Examples
+    --------
+    >>> kge_s = kge(forecasts, obs,preserve_dims='lat')  # if data is of dimension {lat,time}, kge value is computed across the time dimension
+    >>> kge_s = kge(forecasts, obs,reduce_dims="time")  # if data is of dimension {lat,time}, reduce_dims="time" is same as preserve_dims='lat'
+    >>> kge_s = kge(forecasts, obs, return_components=True) # kge_s is dataset of all three components and kge value itself
+    >>> kge_s_weighted = kge(forecasts, obs, scaling_factors=(0.5, 1.0, 2.0))
+
+    References:
+        -   Gupta, H. V., Kling, H., Yilmaz, K. K., & Martinez, G. F. (2009). Decomposition of the mean squared error and
+            NSE performance criteria: Implications for improving hydrological modeling. Journal of Hydrology, 377(1-2), 80-91.
+            https://doi.org/10.1016/j.jhydrol.2009.08.003.
+        -   Knoben, W. J. M., Freer, J. E., & Woods, R. A. (2019). Technical note: Inherent benchmark or not?
+            Comparing Nash-Sutcliffe and Kling-Gupta efficiency scores. Hydrology and Earth System Sciences, 23(10), 4323-4331.
+            https://doi.org/10.5194/hess-23-4323-2019.
+    """
+
+    # Type checks as xrray.corr can only handle xr.DataArray
+    if not isinstance(fcst, xr.DataArray):
+        raise TypeError("kge: fcst must be an xarray.DataArray")
+    if not isinstance(obs, xr.DataArray):
+        raise TypeError("kge: obs must be an xarray.DataArray")
+    if scaling_factors is not None and not isinstance(scaling_factors, (list, np.ndarray)):
+        raise TypeError("kge: scaling_factors must be a list of floats or a numpy array")
```

Comment:
Did you want to check if there are items other than floats in the list when `scaling_factors` is a list? Should it also work if there is an int in the list?

Reply:
It should work with both floats and integers. Since the values must be positive, I also need to check that they are greater than zero.